nixos/modules/services/misc/workout-tracker.nix

```diff
@@ -0,0 +1,121 @@
+  exe = lib.getExe cfg.package;
+in
+
+{
+  options = {
+    services.workout-tracker = {
+      enable = lib.mkEnableOption "A workout tracking web application for personal use (or family, friends), geared towards running and other GPX-based activities";
+
+      package = lib.mkPackageOption pkgs "workout-tracker" { };
+
+      address = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = lib.mdDoc "Web interface address.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8080;
+        description = lib.mdDoc "Web interface port.";
+      };
+
+      stateDir = mkOption {
+        default = "/var/lib/workout-tracker";
+        type = types.str;
+        description = mdDoc "Workout-tracker data directory.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "workout-tracker";
+        description = mdDoc "User account under which Workout-tracker runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "workout-tracker";
+        description = mdDoc "Group under which Workout-tracker runs.";
+      };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/workout-tracker-env";
+        description = mdDoc ''
+          An environment file as defined in {manpage}`systemd.exec(5)`.
+
+          Secrets like `WT_JWT_ENCRYPTION_KEY` may be passed to the service without adding them
+          to the world-readable Nix store.
+        '';
+      };
+
+      settings = mkOption {
+        type = lib.types.submodule {
+          freeformType =
+            with lib.types;
+            attrsOf (
+              let
+                typeList = [
+                  bool
+                  float
+                  int
+                  str
+                  path
+                  package
+                ];
+              in
+              oneOf (
+                typeList
+                ++ [
+                  (listOf (oneOf typeList))
+                  (attrsOf (oneOf typeList))
+                ]
+              )
+            );
+        };
+        default = { };
+        description = lib.mdDoc ''
+          Extra config options.
+        '';
+        example = {
+          WT_LOGGING = "true";
+          WT_DEBUG = "false";
+          WT_DATABASE_DRIVER = "sqlite";
+          WT_DSN = "./database.db";
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.workout-tracker = {
+      description = "A workout tracking web application for personal use (or family, friends), geared towards running and other GPX-based activities";
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        WT_BIND = "${cfg.address}:${toString cfg.port}";
+      };
+      serviceConfig = {
+        ExecStart = "${exe}";
+        WorkingDirectory = cfg.stateDir;
+        User = cfg.user;
```

Comment:
Does this service work with `DynamicUser`? If yes please activate it, if not please add `users.users.${cfg.user}` and `users.groups.${cfg.group}`

Reply:
Shouldn't have any issues with `DynamicUser`. I'm bit low on systemd-unit knowledge in this area and will read more.